test/jdk/javax/swing/Popup/TaskbarPositionTest.java

```diff
@@ -63,27 +87,27 @@ public class TaskbarPositionTest extends JFrame implements ActionListener {
     };
 
     public TaskbarPositionTest() {
-        super("Use CTRL-down to show a JPopupMenu");
-        setContentPane(panel = createContentPane());
-        setJMenuBar(createMenuBar("1 - First Menu", true));
-        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame = new JFrame("Use CTRL-down to show a JPopupMenu");
+        frame.setContentPane(panel = createContentPane());
+        frame.setJMenuBar(createMenuBar("1 - First Menu", true));
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
 
         // CTRL-down will show the popup.
         panel.getInputMap().put(KeyStroke.getKeyStroke(
                 KeyEvent.VK_DOWN, InputEvent.CTRL_MASK), "OPEN_POPUP");
         panel.getActionMap().put("OPEN_POPUP", new PopupHandler());
 
-        pack();
+        frame.pack();
 
         Toolkit toolkit = Toolkit.getDefaultToolkit();
         fullScreenBounds = new Rectangle(new Point(), toolkit.getScreenSize());
         screenBounds = new Rectangle(new Point(), toolkit.getScreenSize());
 
-        // Place the frame near the bottom. This is a pretty wild guess.
-        this.setLocation(0, (int) screenBounds.getHeight() - 2 * this.getHeight());
+        // Place the frame near the bottom.
+        frame.setLocation(0, screenBounds.y + screenBounds.height - frame.getHeight());
```

Comment:
This should be moved below, after the insets of the graphics device are taken into account.